govcd/vdc.go

```diff
@@ -1350,3 +1353,62 @@ func (vdc *Vdc) CloneVapp(sourceVapp *types.CloneVAppParams) (*VApp, error) {
 	err = vapp.Refresh()
 	return vapp, err
 }
+
+// Get the details of a hardware version
+func (vdc *Vdc) GetHardwareVersion(name string) (*types.VirtualHardwareVersion, error) {
+	found := false
+	for _, hwVersion := range vdc.Vdc.Capabilities[0].SupportedHardwareVersions.SupportedHardwareVersion {
```

Comment:
No, not really, the type itself is a bit wonky, hence why we use `Capabilities[0]` instead of just `Capabilities`, but the hardware version support information is always stored in a VDC, so it shouldn't ever be nil.

Reply:
Curious, why it's with a position [0] and can there be unexpected items at other positions?

Reply:
I am in favor of adding a check for `vdc.Capabilities` to contain at least one item. No harm will result from it.

Reply:
> I am in favor of adding a check for vdc.Capabilities to contain at least one item. No harm will result from it.
Agree, added.

> Curious, why it's with a position [0] and can there be unexpected items at other positions?

I would guess he choice to make it a `[]*Capabilities` but not `*Capabilities` came from the name itself. The type's description is `Collection of supported hardware capabilities. [1] ` but I haven't found a way to add more of so-called 'capabilities' to the VDC.

In the API we get the hardware versions using this path:
`[]Capabilities -> SupportedHardwareVersions -> []SupportedHardwareVersion`
even though it technically is something more similar to:
`-> []SupportedHardwareVersions`

[1] - https://developer.vmware.com/apis/1703/vmware-cloud-director/doc/doc/types/CapabilitiesType.html